select.go

```diff
@@ -13,85 +13,155 @@ import (
 	"github.com/manifoldco/promptui/screenbuf"
 )
 
-// SelectedAdd is returned from SelectWithAdd when add is selected.
+// SelectedAdd is used internally inside SelectWithAdd when the add option is selected in select mode.
+// Since -1 is not a possible selected index, this ensure that add mode is always unique inside
+// SelectWithAdd's logic.
 const SelectedAdd = -1
 
-// Select represents a list for selecting a single item
+// Select represents a list of items used to enable selections, they can be used as search engines, menus
+// or as a list of items in a cli based prompt.
+//
+// Basic Usage
```

Comment:
All these examples are great!

Have you seen the godoc example support? it makes them look quite nice in the documentation, imo, and can test/compile them.

for example, check out https://golang.org/src/net/http/example_test.go#L57
which displays as:
https://golang.org/pkg/net/http/#example_FileServer

Reply:
Just saw that. The example I was looking at just used the "ident will format as code" functionality, this will help a lot.

Reply:
I think you can remove this example here, as it's now at the package level.